### PR TITLE
[MLB.tv] New Extractor

### PR DESCRIPTION
**Pull request by ischmidt20**
### Description of your *pull request* and other information

</details>

Adds a new extractor for MLB.tv videos. MLB.tv contains full video and audio replays of every Major League Baseball game. Videos are probably not actually geolocked, but will require a subscription which unfortunately I am unwilling to share.

This pull request is fully "functional," in that it can extract the m3u8 URLs and appropriate headers for each game. However, there are a number of concerns that need to be addressed before merge.

1. Due to the nature of the streams, and limitations with ffmpeg, the issue raised in #4211 means that videos may not download properly. The stream of the actual game is separate from that for the commercials, even though both alternate in the variant m3u8 file. When cleaning up, ffmpeg will only return the stream that starts the playlist. Most of the time, this is the game itself, so the commercials are filtered out, happy surprise. However, if the video starts with commercials, *the whole video will download*, but then the ffmpeg cleanup will result in a video with just commercials. The workaround I used is mentioned at the end of #4211, but it is inelegant and should definitely not be merged into the project.
2. Right now, the "video id" is really a game ID. Each game has multiple broadcasts, usually one video each for the home and away team, along with at least one radio broadcast for each team. Currently all of these broadcasts are returned as different formats, even though the content is of course different for each one. Here is what I mean:

```
ID                         EXT RESOLUTION FPS │   TBR PROTO │ VCODEC        VBR ACODEC    ABR MORE INFO
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Away-en-aac-English_Radio  mp4 audio only     │       m3u8  │ audio only        unknown       [en] English Radio
Away-en-aac-Radio_Española mp4 audio only     │       m3u8  │ audio only        unknown       [es] Radio Española
Home-en-aac-English_Radio  mp4 audio only     │       m3u8  │ audio only        unknown       [en] English Radio
Away-en-50                 mp4 audio only     │   50k m3u8  │ audio only        mp4a.40.2 50k
Away-es-50                 mp4 audio only     │   50k m3u8  │ audio only        mp4a.40.2 50k
Home-en-50                 mp4 audio only     │   50k m3u8  │ audio only        mp4a.40.2 50k
Away-en-153                mp4 320x180     30 │  153k m3u8  │ avc1.4d001f  153k mp4a.40.2  0k
Home-en-153                mp4 320x180     30 │  153k m3u8  │ avc1.4d001f  153k mp4a.40.2  0k
Away-en-620                mp4 384x216     30 │  620k m3u8  │ avc1.4d001f  620k mp4a.40.2  0k
Home-en-620                mp4 384x216     30 │  620k m3u8  │ avc1.4d001f  620k mp4a.40.2  0k
Away-en-960                mp4 512x288     30 │  960k m3u8  │ avc1.4d001f  960k mp4a.40.2  0k
Home-en-960                mp4 512x288     30 │  960k m3u8  │ avc1.4d001f  960k mp4a.40.2  0k
Away-en-1400               mp4 640x360     30 │ 1400k m3u8  │ avc1.4d001f 1400k mp4a.40.2  0k
Home-en-1400               mp4 640x360     30 │ 1400k m3u8  │ avc1.4d001f 1400k mp4a.40.2  0k
Away-en-2120               mp4 896x504     30 │ 2120k m3u8  │ avc1.4d001f 2120k mp4a.40.2  0k
Home-en-2120               mp4 896x504     30 │ 2120k m3u8  │ avc1.4d001f 2120k mp4a.40.2  0k
Away-en-2950               mp4 960x540     30 │ 2950k m3u8  │ avc1.4d001f 2950k mp4a.40.2  0k
Home-en-2950               mp4 960x540     30 │ 2950k m3u8  │ avc1.4d001f 2950k mp4a.40.2  0k
Away-en-4160               mp4 1280x720    30 │ 4160k m3u8  │ avc1.640028 4160k mp4a.40.2  0k
Home-en-4160               mp4 1280x720    30 │ 4160k m3u8  │ avc1.640028 4160k mp4a.40.2  0k
Away-en-6600               mp4 1280x720    60 │ 6600k m3u8  │ avc1.640028 6600k mp4a.40.2  0k
```
It may be better for the extractor to extract at the level of the individual broadcast, rather than the game. However, a key feature of MLB.tv is the ability to play different audio broadcasts that are synced with the selected video, which is why I returned *all* of the broadcasts as a single "video," rather than split by broadcast. This is a question for the prospective users of this extractor and also for the devs, if this violates any hard and fast rules.
3. Currently, the extractor pulls the video by taking in the username (email address) and password for the user's MLB account. However, the program always returns a warning `Login with password is not supported for this website.` Additionally, if no username is supplied, the program just returns `ERROR: quote_from_bytes() expected bytes`. Even though a select few games may be freely available, I would like this extractor to **require** a username and password. Caching the `access_token` might be beneficial as well, but I have no idea how to go about doing that. Dev help would be great in cleaning up the authentication process here.
4. The style is probably really bad. I tried to finesse everything to pass flake8 but it still looks terrible and I'm sure many improvements can be made.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

File: yt_dlp/extractor/mlb.py [hunk 1]
```diff
@@ -263,3 +265,68 @@ def _download_video_data(self, display_id):
...
+    def _perform_login(self, username, password):
+
+        access_token = self._download_json('https://ids.mlb.com/oauth2/aus1m088yK07noBfh356/v1/token', None,
+                                           headers={'User-Agent': 'okhttp/3.12.1',
+                                                    'Content-Type': 'application/x-www-form-urlencoded'},
+                                           data=(('grant_type=password&username=%s&password=%s&scope=openid offline_access'
+                                                  '&client_id=0oa3e1nutA1HLzAKG356') % (urllib.parse.quote(username), urllib.parse.quote(password)))
+                                           .encode())['access_token']
+
+        entitlement = self._download_webpage('https://media-entitlement.mlb.com/api/v3/jwt?os=Android&appname=AtBat&did=' + str(uuid.uuid4()), None,
```
Comment (HobbyistDev) [inline yt_dlp/extractor/mlb.py:294]: I think `os=Android&appname=...` part can moved to `query` field

File: yt_dlp/extractor/mlb.py [hunk 2]
```diff
@@ -263,3 +265,68 @@ def _download_video_data(self, display_id):
...
+
+        return access_token
+
+    def _real_initialize(self):
+        self._access_token = self._perform_login(self.get_param('username'), self.get_param('password'))
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        formats, subtitles = [], []
+        airings = self._download_json(f'https://search-api-mlbtv.mlb.com/svc/search/v2/graphql/persisted/query/core/Airings?variables=%7B%22partnerProgramIds%22%3A%5B%22{video_id}%22%5D%2C%22applyEsniMediaRightsLabels%22%3Atrue%7D', video_id)['data']['Airings']
```
Comment (HobbyistDev) [inline yt_dlp/extractor/mlb.py:313]: Same as above

File: yt_dlp/extractor/mlb.py [hunk 3]
```diff
@@ -263,3 +265,68 @@ def _download_video_data(self, display_id):
   }
 }''' % display_id,
             })['data']['mediaPlayback'][0]
+
+
+class MLBTVIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?mlb\.com/tv/g(?P<id>\d{6})/?.*'
```
Comment (HobbyistDev) [inline yt_dlp/extractor/mlb.py:271]: You forget to define `_NETRC_MATCHINE` variable
From test:
```
AssertionError: None is not true : Extractor MLBTV supports login, but is missing a _NETRC_MACHINE property
```

File: yt_dlp/extractor/mlb.py [hunk 4]
```diff
@@ -263,3 +265,68 @@ def _download_video_data(self, display_id):
...
+                                           headers={'User-Agent': 'okhttp/3.12.1',
+                                                    'Content-Type': 'application/x-www-form-urlencoded'},
+                                           data=(('grant_type=password&username=%s&password=%s&scope=openid offline_access'
+                                                  '&client_id=0oa3e1nutA1HLzAKG356') % (urllib.parse.quote(username), urllib.parse.quote(password)))
+                                           .encode())['access_token']
+
+        entitlement = self._download_webpage('https://media-entitlement.mlb.com/api/v3/jwt?os=Android&appname=AtBat&did=' + str(uuid.uuid4()), None,
+                                             headers={'User-Agent': 'okhttp/3.12.1', 'Authorization': f'Bearer {access_token}'})
+
+        access_token = self._download_json('https://us.edge.bamgrid.com/token', None,
```
Comment (Lesmiscore) [inline yt_dlp/extractor/mlb.py:297]: Assign to `self._access_token` here and remove `_real_initialize`. `self.get_param('username'), self.get_param('password')` may contain credentials for different site

File: yt_dlp/extractor/mlb.py [hunk 5]
```diff
@@ -263,3 +265,68 @@ def _download_video_data(self, display_id):
...
+        },
+    }]
+
+    def _perform_login(self, username, password):
+
+        access_token = self._download_json('https://ids.mlb.com/oauth2/aus1m088yK07noBfh356/v1/token', None,
+                                           headers={'User-Agent': 'okhttp/3.12.1',
+                                                    'Content-Type': 'application/x-www-form-urlencoded'},
+                                           data=(('grant_type=password&username=%s&password=%s&scope=openid offline_access'
+                                                  '&client_id=0oa3e1nutA1HLzAKG356') % (urllib.parse.quote(username), urllib.parse.quote(password)))
```
Comment (Lesmiscore) [inline yt_dlp/extractor/mlb.py:291]: `data` here must be bytes

File: yt_dlp/extractor/mlb.py (hunk 2 above)
Comment (Lesmiscore) [reply yt_dlp/extractor/mlb.py:313]:
```suggestion
        airings = self._download_json(
            f'https://search-api-mlbtv.mlb.com/svc/search/v2/graphql/persisted/query/core/Airings?variables=%7B%22partnerProgramIds%22%3A%5B%22{video_id}%22%5D%2C%22applyEsniMediaRightsLabels%22%3Atrue%7D',
            video_id)['data']['Airings']
```

File: yt_dlp/extractor/mlb.py [hunk 6]
```diff
@@ -263,3 +265,68 @@ def _download_video_data(self, display_id):
...
+                                           .encode())['access_token']
+
+        return access_token
+
+    def _real_initialize(self):
+        self._access_token = self._perform_login(self.get_param('username'), self.get_param('password'))
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        formats, subtitles = [], []
```
Comment (Lesmiscore) [inline yt_dlp/extractor/mlb.py:312]: `formats, subtitles = [], {}`

File: yt_dlp/extractor/mlb.py [hunk 7]
```diff
@@ -263,3 +265,68 @@ def _download_video_data(self, display_id):
...
+        airings = self._download_json(f'https://search-api-mlbtv.mlb.com/svc/search/v2/graphql/persisted/query/core/Airings?variables=%7B%22partnerProgramIds%22%3A%5B%22{video_id}%22%5D%2C%22applyEsniMediaRightsLabels%22%3Atrue%7D', video_id)['data']['Airings']
+        for airing in airings:
+            playback = self._download_json(airing['playbackUrls'][0]['href'].format(scenario='browser~csai'), video_id,
+                                           headers={'Authorization': self._access_token,
+                                                    'Accept': 'application/vnd.media-service+json; version=2'})
+            m3u8_url = playback['stream']['complete']
+            f, s = self._extract_m3u8_formats_and_subtitles(m3u8_url, video_id, 'mp4', m3u8_id=airing['feedType'] + '-' + airing['feedLanguage'])
+            formats.extend(f)
+            subtitles.extend(s)
```
Comment (Lesmiscore) [inline yt_dlp/extractor/mlb.py:321]: `self._merge_subtitles(s, subtitles)`

**Review by Lesmiscore — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/mlb.py (hunk 3 above)
Comment (pukkandan) [inline yt_dlp/extractor/mlb.py:271]:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?mlb\.com/tv/g(?P<id>\d{6})'
```

File: yt_dlp/extractor/mlb.py [hunk 8]
```diff
@@ -263,3 +265,68 @@ def _download_video_data(self, display_id):
...
+            title = airing['titles'][0]['episodeName']
```
Comment (pukkandan) [inline yt_dlp/extractor/mlb.py:322]: make non-fatal (use `traverse_obj`)

File: yt_dlp/extractor/mlb.py [hunk 9]
```diff
@@ -263,3 +265,68 @@ def _download_video_data(self, display_id):
...
+
+        self._sort_formats(formats)
+
+        return {
+            'title': title,
+            'formats': formats,
+            'subtitles': subtitles,
+            'http_headers': {'Authorization': 'Bearer ' + self._access_token},
```
Comment (pukkandan) [inline yt_dlp/extractor/mlb.py:330]: Use fstrings, same above

File: yt_dlp/extractor/mlb.py [hunk 10]
```diff
@@ -263,3 +265,68 @@ def _download_video_data(self, display_id):
...
+        self._access_token = self._perform_login(self.get_param('username'), self.get_param('password'))
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        formats, subtitles = [], []
+        airings = self._download_json(f'https://search-api-mlbtv.mlb.com/svc/search/v2/graphql/persisted/query/core/Airings?variables=%7B%22partnerProgramIds%22%3A%5B%22{video_id}%22%5D%2C%22applyEsniMediaRightsLabels%22%3Atrue%7D', video_id)['data']['Airings']
+        for airing in airings:
+            playback = self._download_json(airing['playbackUrls'][0]['href'].format(scenario='browser~csai'), video_id,
+                                           headers={'Authorization': self._access_token,
+                                                    'Accept': 'application/vnd.media-service+json; version=2'})
```
Comment (pukkandan) [inline yt_dlp/extractor/mlb.py:317]: Have you read the style guide? similar changes elsewhere
```suggestion
            playback = self._download_json(
                airing['playbackUrls'][0]['href'].format(scenario='browser~csai'),
                 video_id, headers={
                     'Authorization': self._access_token,
                     'Accept': 'application/vnd.media-service+json; version=2'
                 })
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

**Comment by pukkandan** [comment]
I'm a bit confused about the first point. First off, could you clarify whether these are live streams? or VODs? (If live, you need return `'is_live': True`)

**Comment by ischmidt20** [comment]
> I'm a bit confused about the first point. First off, could you clarify whether these are live streams? or VODs? (If live, you need return `'is_live': True`)

MLB.tv contains both live streams and VODs. I've primarily written this extractor for the VODs, but I've just thrown a live game in there, and the formats seem to be extracted in the same way. There probably is a way for me to determine `is_live` from some API call.

File: yt_dlp/extractor/mlb.py (hunk 5 above)
Comment (ischmidt20) [reply yt_dlp/extractor/mlb.py:291]: The string is encoded as bytes on the following line

**Comment by pukkandan** [comment]
In that case, https://github.com/yt-dlp/yt-dlp/issues/4211#issuecomment-1168169087 is an acceptable solution for the time being

**Comment by ischmidt20** [comment]
I should note that the strange behavior only appears for some videos. It was happening a lot when I posted that original issue back in June, but I haven't seen it so far on anything that I've tested this extractor on so far. Will continue to investigate this

File: yt_dlp/extractor/mlb.py (hunk 3 above)
Comment (ischmidt20) [reply yt_dlp/extractor/mlb.py:272]: Neat, didn't know we don't need to match the whole UR:

File: yt_dlp/extractor/mlb.py (hunk 4 above)
Comment (ischmidt20) [reply yt_dlp/extractor/mlb.py:299]: Thanks, this is what I was needing help with!

File: yt_dlp/extractor/mlb.py (hunk 10 above)
Comment (ischmidt20) [reply yt_dlp/extractor/mlb.py]: How is the style now?

File: yt_dlp/extractor/mlb.py (hunk 1 above)
Comment (ischmidt20) [reply yt_dlp/extractor/mlb.py]: I am not quite sure what you mean here... is `query` preferred to just using the URL?

**Review by pukkandan — CHANGES_REQUESTED** [review]
> I should note that the strange behavior only appears for some videos. It was happening a lot when I posted that original issue back in June, but I haven't seen it so far on anything that I've tested this extractor on so far. Will continue to investigate this

Should I merge without addressing this? or wait for you to investigate?

**Comment by ischmidt20** [comment]
As the potential fixes would require changes external the extractor, I think it is fine to merge once I get these style commits in

**Comment by ischmidt20** [comment]
Changes implemented

File: yt_dlp/extractor/mlb.py [hunk 11]
```diff
@@ -263,3 +267,75 @@ def _download_video_data(self, display_id):
...
+        data = f'grant_type=urn:ietf:params:oauth:grant-type:token-exchange&subject_token={entitlement}&subject_token_type=urn:ietf:params:oauth:token-type:jwt&platform=android-tv'
+        access_token = self._download_json(
+            'https://us.edge.bamgrid.com/token', None,
+            headers={
+                'Accept': 'application/json',
+                'Authorization': 'Bearer bWxidHYmYW5kcm9pZCYxLjAuMA.6LZMbH2r--rbXcgEabaDdIslpo4RyZrlVfWZhsAgXIk',
+                'Content-Type': 'application/x-www-form-urlencoded'
+            }, data=data.encode())['access_token']
+
+        self._access_token = access_token
```
Comment (pukkandan) [inline yt_dlp/extractor/mlb.py:313]:
```suggestion
        self._access_token = self._download_json(
            'https://us.edge.bamgrid.com/token', None,
            headers={
                'Accept': 'application/json',
                'Authorization': 'Bearer bWxidHYmYW5kcm9pZCYxLjAuMA.6LZMbH2r--rbXcgEabaDdIslpo4RyZrlVfWZhsAgXIk',
                'Content-Type': 'application/x-www-form-urlencoded'
            }, data=data.encode())['access_token']
```

**Review by pukkandan — APPROVED** [review]

File: yt_dlp/extractor/mlb.py [hunk 12]
```diff
@@ -263,3 +267,73 @@ def _download_video_data(self, display_id):
...
+        'info_dict': {
+            'id': '661581',
+            'ext': 'mp4',
+            'title': '2022-07-02 - St. Louis Cardinals @ Philadelphia Phillies',
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }]
+
```
Comment (pukkandan) [inline yt_dlp/extractor/mlb.py:287]: All video on the site needs account, correct? Do this to give users a proper error message if try try without `-u`.
```suggestion
    _access_token = None
    
    def _real_initialize(self):
        if not self._access_token:
            self.raise_login_required(
                'All videos are only available to registered users', method='password')
```

Comment (ischmidt20) [reply yt_dlp/extractor/mlb.py:287]: There are a small fraction of videos that are "free" (no paid subscription required) but I'm not sure if they still require an account or not. Either way, I am fine requiring an account for any video in the service.

Comment (ischmidt20) [reply yt_dlp/extractor/mlb.py:293]: Just verified that a free video still requires an account

File: yt_dlp/extractor/mlb.py [hunk 13]
```diff
@@ -263,3 +267,79 @@ def _download_video_data(self, display_id):
...
+                'Authorization': 'Bearer bWxidHYmYW5kcm9pZCYxLjAuMA.6LZMbH2r--rbXcgEabaDdIslpo4RyZrlVfWZhsAgXIk',
+                'Content-Type': 'application/x-www-form-urlencoded'
+            }, data=data.encode())['access_token']
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        formats, subtitles = [], {}
+        airings = self._download_json(
+            f'https://search-api-mlbtv.mlb.com/svc/search/v2/graphql/persisted/query/core/Airings?variables=%7B%22partnerProgramIds%22%3A%5B%22{video_id}%22%5D%2C%22applyEsniMediaRightsLabels%22%3Atrue%7D',
+            video_id)['data']['Airings']
```
Comment (pukkandan) [inline yt_dlp/extractor/mlb.py:324]:
```suggestion
        airings = self._download_json(
            f'https://search-api-mlbtv.mlb.com/svc/search/v2/graphql/persisted/query/core/Airings?variables=%7B%22partnerProgramIds%22%3A%5B%22{video_id}%22%5D%2C%22applyEsniMediaRightsLabels%22%3Atrue%7D',
            video_id)['data']['Airings']
    
        formats, subtitles = [], {}
```

File: yt_dlp/extractor/mlb.py [hunk 14]
```diff
@@ -263,3 +267,79 @@ def _download_video_data(self, display_id):
...
+        for airing in airings:
+            playback = self._download_json(
+                airing['playbackUrls'][0]['href'].format(scenario='browser~csai'), video_id,
+                headers={
+                    'Authorization': self._access_token,
+                    'Accept': 'application/vnd.media-service+json; version=2'
+                })
+            m3u8_url = playback['stream']['complete']
+            f, s = self._extract_m3u8_formats_and_subtitles(
```
Comment (pukkandan) [inline yt_dlp/extractor/mlb.py:333]:
```suggestion
            m3u8_url = self._download_json(
                airing['playbackUrls'][0]['href'].format(scenario='browser~csai'), video_id,
                headers={
                    'Authorization': self._access_token,
                    'Accept': 'application/vnd.media-service+json; version=2'
                })['stream']['complete']
            f, s = self._extract_m3u8_formats_and_subtitles(
```

File: yt_dlp/extractor/mlb.py [hunk 15]
```diff
@@ -263,3 +267,79 @@ def _download_video_data(self, display_id):
...
+            formats.extend(f)
+            self._merge_subtitles(s, subtitles)
+
+        self._sort_formats(formats)
+        return {
+            'title': traverse_obj(airings, (..., 'titles', 0, 'episodeName'), get_all=False),
+            'formats': formats,
+            'subtitles': subtitles,
+            'http_headers': {'Authorization': f'Bearer {self._access_token}'},
+            'id': video_id
```
Comment (pukkandan) [inline yt_dlp/extractor/mlb.py:344]:
```suggestion
            'id': video_id,
            'title': traverse_obj(airings, (..., 'titles', 0, 'episodeName'), get_all=False),
            'formats': formats,
            'subtitles': subtitles,
            'http_headers': {'Authorization': f'Bearer {self._access_token}'},
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/mlb.py [hunk 16]
```diff
@@ -263,3 +267,79 @@ def _download_video_data(self, display_id):
...
+            m3u8_url = self._download_json(
+                airing['playbackUrls'][0]['href'].format(scenario='browser~csai'), video_id,
+                headers={
+                    'Authorization': self._access_token,
+                    'Accept': 'application/vnd.media-service+json; version=2'
+                })['stream']['complete']
+            f, s = self._extract_m3u8_formats_and_subtitles(
+                m3u8_url, video_id, 'mp4', m3u8_id=join_nonempty(airing.get('feedType'), airing.get('feedLanguage')))
+            formats.extend(f)
+            self._merge_subtitles(s, subtitles)
```
Comment (pukkandan) [inline yt_dlp/extractor/mlb.py:336]:
```suggestion
            self._merge_subtitles(s, target=subtitles)
```

**Pull request merged**